powertools-idempotency/src/main/java/software/amazon/lambda/powertools/idempotency/persistence/DynamoDBPersistenceStore.java

```diff
@@ -90,13 +94,14 @@ private DynamoDBPersistenceStore(String tableName,
             if (idempotencyDisabledEnv == null || idempotencyDisabledEnv.equalsIgnoreCase("false")) {
                 DynamoDbClientBuilder ddbBuilder = DynamoDbClient.builder()
                         .httpClient(UrlConnectionHttpClient.builder().build())
+                        .overrideConfiguration(ClientOverrideConfiguration.builder().putAdvancedOption(SdkAdvancedClientOption.USER_AGENT_SUFFIX, UserAgentConfigurer.getUserAgent(IDEMPOTENCY)).build())
                         .region(Region.of(System.getenv(AWS_REGION_ENV)));
 
                 // AWS_LAMBDA_INITIALIZATION_TYPE has two values on-demand and snap-start
                 // when using snap-start mode, the env var creds provider isn't used and causes a fatal error if set
                 // fall back to the default provider chain if the mode is anything other than on-demand.
                 String initializationType = System.getenv().get(AWS_LAMBDA_INITIALIZATION_TYPE);
-                if (initializationType  != null && initializationType.equals(ON_DEMAND)) {
+                if (initializationType != null && initializationType.equals(ON_DEMAND)) {
```

Comment:
I see this is just a syntax change, but we are about to change the credential chain as part of #1303 and we'll need to keep on the two merges.

Reply:
FYI - #1303 has been merged to main; if you merge down now you'll get this change